integration/tests/access/unstaked_node_test.go

```diff
@@ -0,0 +1,131 @@
+package access
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/onflow/flow-go/cmd/bootstrap/utils"
+	"github.com/onflow/flow-go/crypto"
+	consensus_follower "github.com/onflow/flow-go/follower"
+	"github.com/onflow/flow-go/integration/testnet"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+type UnstakedAccessSuite struct {
+	suite.Suite
+
+	// root context for the current test
+	ctx    context.Context
+	cancel context.CancelFunc
+
+	net        *testnet.FlowNetwork
+	stakedID   flow.Identifier
+	unstakedID flow.Identifier
+	conID      flow.Identifier
+	follower   consensus_follower.ConsensusFollower
+}
+
+func TestUnstakedAccessSuite(t *testing.T) {
+	suite.Run(t, new(UnstakedAccessSuite))
+}
+
+func (suite *UnstakedAccessSuite) TearDownTest() {
+	// avoid nil pointer errors for skipped tests
+	if suite.cancel != nil {
+		defer suite.cancel()
+	}
+	if suite.net != nil {
+		suite.net.Remove()
+	}
+}
+
+func (suite *UnstakedAccessSuite) SetupTest() {
+	suite.buildNetworkConfig()
+	// start the network
+	suite.ctx, suite.cancel = context.WithCancel(context.Background())
+	suite.net.Start(suite.ctx)
+}
+
+func (suite *UnstakedAccessSuite) TestReceiveBlocks() {
+	go suite.follower.Run(suite.ctx)
+	// TODO: to be implemented later
+	time.Sleep(time.Second * 30)
+}
+
+func (suite *UnstakedAccessSuite) OnBlockFinalizedConsumer(finalizedBlockID flow.Identifier) {
+	fmt.Println(finalizedBlockID.String())
+}
+
+func (suite *UnstakedAccessSuite) buildNetworkConfig() {
+
+	// staked access node
+	suite.stakedID = unittest.IdentifierFixture()
+	stakedConfig := testnet.NewNodeConfig(
+		flow.RoleAccess,
+		testnet.WithID(suite.stakedID),
+		testnet.SupportsUnstakedNodes(),
+		testnet.WithLogLevel(zerolog.TraceLevel),
+	)
+
+	collectionConfigs := []func(*testnet.NodeConfig){
+		testnet.WithAdditionalFlag("--hotstuff-timeout=12s"),
+		testnet.WithAdditionalFlag("--block-rate-delay=100ms"),
+		testnet.WithLogLevel(zerolog.WarnLevel),
+		// TODO replace these with actual values
+		testnet.WithAdditionalFlag("--access-address=null"),
```

Comment:
what is this flag?

Reply:
removed it - it is basically a setting to allow the collection node to talk to the access node (needed for epoch)